translation.rst

```diff
@@ -160,27 +157,121 @@ Now, if the language of the user's locale is French (e.g. ``fr_FR`` or ``fr_BE``
+    avoid repeating yourself if you use keywords instead of real text for your
+    ids:
+
+    .. configuration-block::
+
+        .. code-block:: yaml
+
+            symfony:
+                is:
+                    # id is symfony.is.great
+                    great: Symfony is great
+                    # id is symfony.is.amazing
+                    amazing: Symfony is amazing
+                has:
+                    # id is symfony.has.bundles
+                    bundles: Symfony has bundles
+            user:
+                # id is user.login
+                login: Login
+
+        .. code-block:: php
+
+            [
+                'symfony' => [
+                    'is' => [
+                        // id is symfony.is.great
+                        'great'   => 'Symfony is great',
+                        // id is symfony.is.amazing
+                        'amazing' => 'Symfony is amazing',
+                    ],
+                    'has' => [
+                        // id is symfony.has.bundles
+                        'bundles' => 'Symfony has bundles',
+                    ],
+                ],
+                'user' => [
+                    // id is user.login
+                    'login' => 'Login',
+                ],
+            ];
+
 The Translation Process
 ~~~~~~~~~~~~~~~~~~~~~~~
 
 To actually translate the message, Symfony uses a simple process:
 
-* The ``locale`` of the current user, which is stored on the request is determined;
+#. The ``locale`` of the current user, which is stored on the request is determined;
 
-* A catalog (e.g. big collection) of translated messages is loaded from translation
-  resources defined for the ``locale`` (e.g. ``fr_FR``). Messages from the
-  :ref:`fallback locale <translation-fallback>` are also loaded and
-  added to the catalog if they don't already exist. The end result is a large
-  "dictionary" of translations.
+#. A catalog (i.e. big collection) of translated messages is loaded from translation
+   resources defined for the ``locale`` (e.g. ``fr_FR``). Messages from the
+   :ref:`fallback locale <translation-fallback>` are also loaded and
+   added to the catalog if they don't already exist. The end result is a large
+   "dictionary" of translations.
 
-* If the message is located in the catalog, the translation is returned. If
-  not, the translator returns the original message.
+#. If the message is located in the catalog, the translation is returned. If
+   not, the translator returns the original message.
 
 When using the ``trans()`` method, Symfony looks for the exact string inside
 the appropriate message catalog and returns it (if it exists).
 
+.. _using-message-domains:
+
+Using Message Domains
+~~~~~~~~~~~~~~~~~~~~~
+
+As you've seen, message files are organized into the different locales that
+they translate. The message files can also be organized further into "domains".
+
+The domain is the filename without the extension. The default domain is
```

Comment:
`without the extension` -> `without any extensions` ?  `messages.en.xlf`  both `en` and `xlf` are extensions .. although `en` is a special meaning extension.